Gems/ROS2/Code/Source/RobotImporter/SDFormat/Hooks/ROS2LidarSensorHook.cpp

```diff
@@ -51,28 +52,50 @@ namespace ROS2::SDFormat
+
             SensorConfiguration sensorConfiguration;
             sensorConfiguration.m_frequency = sdfSensor.UpdateRate();
-            const AZStd::string messageType = "sensor_msgs::msg::PointCloud2";
-            Utils::AddTopicConfiguration(sensorConfiguration, "pc", messageType, messageType);
+            const AZStd::string messageType = is2DLidar ? "sensor_msgs::msg::LaserScan" : "sensor_msgs::msg::PointCloud2";
+            const AZStd::string messageTopic = is2DLidar ? "scan" : "pc";
+            Utils::AddTopicConfiguration(sensorConfiguration, messageTopic, messageType, messageType);
 
-            LidarSensorConfiguration lidarConfiguration;
-            lidarConfiguration.m_lidarParameters.m_model = LidarTemplate::LidarModel::Custom3DLidar;
+            LidarSensorConfiguration lidarConfiguration{ is2DLidar ? LidarTemplateUtils::Get2DModels()
+                                                                   : LidarTemplateUtils::Get3DModels() };
             lidarConfiguration.m_lidarParameters.m_name = AZStd::string(sdfSensor.Name().c_str());
             lidarConfiguration.m_lidarParameters.m_minHAngle = lidarSensor->HorizontalScanMinAngle().Degree();
             lidarConfiguration.m_lidarParameters.m_maxHAngle = lidarSensor->HorizontalScanMaxAngle().Degree();
             lidarConfiguration.m_lidarParameters.m_minVAngle = lidarSensor->VerticalScanMinAngle().Degree();
             lidarConfiguration.m_lidarParameters.m_maxVAngle = lidarSensor->VerticalScanMaxAngle().Degree();
-            lidarConfiguration.m_lidarParameters.m_layers = lidarSensor->HorizontalScanSamples();
-            lidarConfiguration.m_lidarParameters.m_numberOfIncrements = lidarSensor->VerticalScanSamples();
+            lidarConfiguration.m_lidarParameters.m_numberOfIncrements = lidarSensor->HorizontalScanSamples();
+            lidarConfiguration.m_lidarParameters.m_layers = lidarSensor->VerticalScanSamples();
             lidarConfiguration.m_lidarParameters.m_minRange = lidarSensor->RangeMin();
             lidarConfiguration.m_lidarParameters.m_maxRange = lidarSensor->RangeMax();
 
+            const auto lidarSystems = LidarRegistrarInterface::Get()->GetRegisteredLidarSystems();
+            if (!lidarSystems.empty())
+            {
+                const AZStd::string query = (sdfSensor.Type() == sdf::SensorType::GPU_LIDAR ? "RobotecGPULidar" : LidarSystem::SystemName);
+                if (const auto it = AZStd::find(lidarSystems.begin(), lidarSystems.end(), query); it != lidarSystems.end())
+                {
+                    lidarConfiguration.m_lidarSystem = *it;
+                }
+            }
+
+            AZ_Warning("ROS2LidarSensorHook", !lidarConfiguration.m_lidarSystem.empty(), "Lidar System not set.");
+            if (sdfSensor.Type() == sdf::SensorType::GPU_LIDAR)
+            {
+                AZ_Info("ROS2LidarSensorHook", "Enable RGL Gem to import GPU Lidar.");
```

Comment:
Escalate this to at least a warning please. 
What are the user steps to enable or to switch from what was mapped to what works? 
Please describe what they should do.

Reply:
Escalated to warning and added RGL link.